lib/fizzy/leb128.hpp

```diff
@@ -33,15 +33,15 @@ std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input)
 
     using T_unsigned = typename std::make_unsigned<T>::type;
     T_unsigned result = 0;
-    int result_shift = 0;
+    size_t result_shift = 0;
 
     for (; result_shift < std::numeric_limits<T_unsigned>::digits; ++input, result_shift += 7)
     {
         result |= static_cast<T_unsigned>((static_cast<T_unsigned>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
         {
             // sign extend
-            if ((*input & 0x40) != 0)
+            if ((*input & 0x40) != 0 && result_shift + 7 < sizeof(T_unsigned) * 8)
```

Comment:
```suggestion
            if ((*input & 0x40) != 0 && (result_shift + 7 < sizeof(T_unsigned) * 8))
```

I know it is not needed by improves readability.

Reply:
Added